examples/ComplexExample/package.json

```diff
@@ -0,0 +1,29 @@
+{
+  "name": "RNExperimental",
+  "version": "0.0.1",
+  "private": true,
+  "scripts": {
+    "start": "node node_modules/react-native/local-cli/cli.js start",
+    "test": "jest"
+  },
+  "dependencies": {
+    "react": "15.4.2",
+    "react-native": "0.40.0",
+    "react-native-vector-icons": "^4.0.0",
+    "react-navigation": "git+https://github.com/react-community/react-navigation.git",
```

Comment:
I always run into this `react-native-vector-icons` issue when I have a forked package that also has that dependency. As long as you can make sure the react-native packager is not picking up that vector package twice, the error won't come up. 

One way (which I'm not sure applies in your setup) for me was to make my own blacklist file: `rn-cli.config.js` and replace npm start script with `node node_modules/react-native/local-cli/cli.js start  --config=rn-cli.config.js`. This way I made sure my forked package wouldn't be picked up by the packager.

Here is more info about it: https://github.com/facebook/react-native/issues/7271

My blacklist file sample (RN 0.43.4, react-navigation 1.0.0-beta.9):
```
const blacklist = require('react-native/packager/blacklist');

console.log('blacklisting folder: forked_packages/...');

module.exports = {
    getBlacklistRE: () => blacklist([
        /forked_packages\/.*/,
    ])
};
```